src/contexts/MetaContext.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { createContext, PropsWithChildren, useContext } from 'react';
+import { AppPluginMeta } from '@grafana/data';
+
+import type { GlobalSettings } from 'types';
+
+interface VerifiedMeta extends Omit<AppPluginMeta<GlobalSettings>, 'jsonData'> {
+  jsonData: GlobalSettings;
+}
+
+type MetaContextValue = {
+  meta: VerifiedMeta;
+} | null;
+
+export const MetaContext = createContext<MetaContextValue>(null);
+
+interface MetaContextProviderProps extends PropsWithChildren {
+  meta: AppPluginMeta<GlobalSettings>;
+}
+
+export const MetaContextProvider = ({ children, meta }: MetaContextProviderProps) => {
+  const verifiedMeta = verifyMeta(meta);
+
+  return <MetaContext.Provider value={{ meta: verifiedMeta }}>{children}</MetaContext.Provider>;
+};
+
+export function useMetaContext() {
+  const context = useContext(MetaContext);
+
+  if (!context) {
+    throw new Error('useMetaContext must be used within a MetaContextProvider');
+  }
+
+  return context;
+}
+
+function verifyMeta(meta: AppPluginMeta<GlobalSettings>): VerifiedMeta {
+  // if (!meta.jsonData) {
+  //   console.log(meta);
+  //   throw new Error('verifyMeta: meta.jsonData is required');
+  // }
+
+  // if (!meta.jsonData.stackId) {
+  //   console.log(meta);
+  //   throw new Error('verifyMeta: meta.jsonData.stackId is required');
+  // }
```

Comment:
Should this be uncommented?

Reply:
I've removed the commented out code and left a todo comment. I want to circle back to this problem in a future PR because there's a bit of a problem with how we share components from the Plugin to the Datasource and what assumptions they expect of data to be available.

The Datasource provides us the meta data for the Datasource plugin and the App plugin provides us the meta data for App Plugin and they aren't 1:1 so we need to have some sort of normalisation for components they want to share and not just assume they are the same. This is especially true when the state of provisioned / initialised gets involved.

To go someway to addressing this I've removed the `<LinkedDatasource />` components we use in the datasource's `<ConfigEditor />` as they expect plugin meta data. I'll add something back at a later date but I want to address the problem of unifying where we read the logs / metrics datasources from because at the moment it is difficult to reason with.